Extensions/PathfindingBehavior/tests/legacypathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,688 @@
+        obstacleX = playerX;
+        obstacleY = playerY + playerBottomBorder + obstacleTopBorder;
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move away from the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(3);
+  });
+
+  it('can find a path when an obstacle is overlaping only at the start', function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        // The obstacle will be right on the player right after the first step.
+        // So, it's overlaping the player by one cell lengt at start.
+        obstacleX =
+          playerX +
+          playerRightBorder +
+          obstacleLeftBorder -
+          cellSize +
+          inclusiveMargin;
+        obstacleY = playerY;
+        targetX = playerX - 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'left':
+        obstacleX =
+          playerX -
+          playerLeftBorder -
+          obstacleRightBorder +
+          cellSize -
+          inclusiveMargin;
+        obstacleY = playerY;
+        targetX = playerX + 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'up':
+        obstacleX = playerX;
+        obstacleY =
+          playerY -
+          playerTopBorder -
+          obstacleBottomBorder +
+          cellSize -
+          inclusiveMargin;
+        targetX = playerX;
+        targetY = playerY + 2 * cellSize;
+        break;
+      case 'down':
+        obstacleX = playerX;
+        obstacleY =
+          playerY +
+          playerBottomBorder +
+          obstacleTopBorder -
+          cellSize +
+          inclusiveMargin;
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move away from the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(3);
+  });
+
+  it("mustn't find a path when an obstacle is slightly overlaping the only first cell to go", function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        // The obstacle will be right on the player right after the first step.
+        // So, it's overlaping the player by one cell lengt at start.
```

Comment:
lengt: length
overlaping: overlapping

Reply:
(here and in the rest of the code)

Reply:
I installed a spelling checker extension on VS. It will be very helpful.